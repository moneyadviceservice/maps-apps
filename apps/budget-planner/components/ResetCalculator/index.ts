import ResetCalculator from './ResetCalculator';
export type { ResetCalculatorProps } from './ResetCalculator';
export default ResetCalculator;
