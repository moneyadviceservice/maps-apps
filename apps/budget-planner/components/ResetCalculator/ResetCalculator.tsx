import { SyntheticEvent } from 'react';

import { TranslationGroup } from 'data/types';

import { Button } from '@maps-react/common/components/Button';
import { H2 } from '@maps-react/common/components/Heading';
import useTranslation from '@maps-react/hooks/useTranslation';

export type ResetCalculatorProps = {
  title: TranslationGroup;
  description: TranslationGroup;
  confirmLabel: TranslationGroup;
  cancelLabel: TranslationGroup;
  confirmAction: string;
  cancelAction: string;
  onCancelClick?: (event: SyntheticEvent) => void;
};

const ResetCalculator = ({
  title,
  description,
  confirmLabel,
  cancelLabel,
  confirmAction,
  cancelAction,
  onCancelClick,
}: ResetCalculatorProps) => {
  const { z } = useTranslation();

  return (
    <div id="ajax">
      <H2 color="text-blue-800">{z(title)}</H2>
      <p className="pb-4">{z(description)}</p>
      <div className="flex flex-row items-start gap-5">
        <Button formAction={confirmAction}>{z(confirmLabel)}</Button>
        <Button
          variant="secondary"
          formAction={cancelAction}
          onClick={onCancelClick}
        >
          {z(cancelLabel)}
        </Button>
      </div>
    </div>
  );
};

export default ResetCalculator;
