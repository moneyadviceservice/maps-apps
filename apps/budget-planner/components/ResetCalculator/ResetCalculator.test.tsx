import { render } from '@testing-library/react';
import ResetCalculator from '../ResetCalculator';

jest.mock('@maps-react/hooks/useTranslation', () => {
  return jest.fn(() => ({
    z: (t: { en: string; cy: string }) => t.en,
  }));
});

const defaultProps = {
  title: {
    en: 'Reset the calculator',
    cy: 'Ailosod y gyfrifiannell',
  },
  description: {
    en: 'Are you sure you want to clear your budget information?',
    cy: "Ydych chi'n siÅµr eich bod am glirio'ch holl wybodaeth Gyllideb?",
  },
  confirmLabel: { en: 'Confirm', cy: 'Cadarnhau' },
  cancelLabel: { en: 'Cancel', cy: 'Canslo' },
  confirmAction: '/api/reset-calculator',
  cancelAction: 'api/cancel',
  onCancelClick: () => console.log('Cancel clicked'),
};

describe('ResetCalculator component', () => {
  it('renders correctly', () => {
    const { container } = render(<ResetCalculator {...defaultProps} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
