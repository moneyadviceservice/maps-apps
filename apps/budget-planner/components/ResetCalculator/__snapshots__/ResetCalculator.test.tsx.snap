// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetCalculator component renders correctly 1`] = `
<div
  id="ajax"
>
  <h2
    class="text-3xl md:text-4xl text-blue-800 font-bold"
  >
    Reset the calculator
  </h2>
  <p
    class="pb-4"
  >
    Are you sure you want to clear your budget information?
  </p>
  <div
    class="flex flex-row items-start gap-5"
  >
    <button
      class="border-0 text-white bg-pink-600 hover:bg-pink-800 hover:shadow-none active:bg-pink-900 active:outline-yellow-200 active:text-white t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-primary-button"
      formaction="/api/reset-calculator"
      type="submit"
    >
      <span />
      Confirm
      <span />
    </button>
    <button
      class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-secondary-button"
      formaction="api/cancel"
      type="submit"
    >
      <span />
      Cancel
      <span />
    </button>
  </div>
</div>
`;
